lib/daos/subjectsDao.js

```diff
@@ -0,0 +1,61 @@
+import { subjects, users } from 'models';
+import sequelize from 'sequelize';
+import { badRequest } from 'utils/responseInterceptors';
+
+const attributes = ['id', 'name'];
+export const findOneSubject = id =>
+    subjects.findOne({ where: { id }, attributes });
+
+export const findAllSubjects = async (where, page, limit) => {
+    const totalCount = await subjects.count({ where });
+    const allSubjects = await subjects.findAll({
+        where,
+        attributes,
+        order: sequelize.literal('id ASC'),
+        limit,
+        offset: (page - 1) * limit,
+        include: [
+            {
+                model: users,
+                attributes: ['firstName']
+            }
+        ]
+    });
+    return {
+        allSubjects,
+        totalCount
+    };
+};
+
+export const updateOneSubject = async (id, subject) => {
+    await subjects.update(subject, {
+        where: {
+            id
+        }
+    });
+    const get = await findOneSubject(id);
```

Comment:
let's please rename

Reply:
this would be more appropriate. 
Trying to update something that doesn't exist plus looking it up isn't ideal
```
export const updateOneSubject = async (id, updatedSubject) => {
    const subject = await findOneSubject(id);
    if (subject) {
        await subject.update(updatedSubject);
        return { name: subject.name, id: subject.id };
    } else {
        return badRequest('Did not update anything, please check the Id');
    }
};
```